### PR TITLE
feat: add cheque-limit parameter to extrinsic execute

### PR DESCRIPTION
**Pull request by neutrinoks**

### CODE REVIEW

**Review by Rqnsom — DISMISSED** [review]
LGTM

File: tests/balance.rs [hunk 1]
```diff
@@ -264,16 +271,61 @@ fn verify_insufficient_balance() {
         assert!(MoveModule::execute(
             RuntimeOrigin::signed(alice_addr_32.clone()),
             transaction_bc,
-            AMOUNT as u64,
+            MAX_GAS_AMOUNT,
```
[the remark this answers is not in the document]
Comment (neutrinoks) [reply tests/balance.rs:266]: Done :+1:

File: tests/balance.rs [hunk 2]
```diff
@@ -264,16 +271,61 @@ fn verify_insufficient_balance() {
...
 fn verify_move_script_fails_after_successful_transfer() {
+    const BALANCE: BalanceOf<Test> = 1000;
+    const AMOUNT: BalanceOf<Test> = 100;
+
+    let (alice_addr_32, alice_addr_mv) = addrs_from_ss58(ALICE_ADDR).unwrap();
+    let (bob_addr_32, bob_addr_mv) = addrs_from_ss58(BOB_ADDR).unwrap();
+
     new_test_ext().execute_with(|| {
-        unimplemented!();
+        // Set Alice's and Bob's balances to a predefined value
```
[the remark this answers is not in the document]
Comment (neutrinoks) [reply tests/balance.rs]: Done

File: tests/balance.rs [hunk 3]
```diff
@@ -264,16 +271,61 @@ fn verify_insufficient_balance() {
...
+            MAX_GAS_AMOUNT,
+            AMOUNT,
+        )
+        .is_err());
+
+        // Verify balances have not been modified and transfer was not applied.
+        let now_blnc_alice = Balances::free_balance(&alice_addr_32);
+        let now_blnc_bob = Balances::free_balance(&bob_addr_32);
+        assert_eq!(now_blnc_alice, BALANCE);
+        assert_eq!(now_blnc_bob, BALANCE);
```
[the remark this answers is not in the document]
Comment (neutrinoks) [reply tests/balance.rs:321]: In this PR or in another one? :wink:

[the remark this answers is not in the document]
Comment (Rqnsom) [reply tests/balance.rs:321]: Let's do it in this one - I'm not sure we'll have more balance-related PRs after this one?

[the remark this answers is not in the document]
Comment (neutrinoks) [reply tests/balance.rs:321]: Done

**Review by Rqnsom — APPROVED** [review]
👍

**Pull request merged**